reader-writer-lock/src/main/java/com/iluwatar/reader/writer/lock/ReaderWriterLock.java

```diff
@@ -0,0 +1,147 @@
+package com.iluwatar.reader.writer.lock;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Class responsible for control the access for reader or writer
```

Comment:
Javadoc can be a bit more detailed here. "Allows multiple readers to hold the lock at same time, but if any writer holds the lock then readers wait. If reader holds the lock then writer waits." This lock is _not fair_.